### PR TITLE
Add a new update-credential command

### PR DESCRIPTION
**Pull request by wallyworld**
Fixes: https://bugs.launchpad.net/juju/+bug/1623811

A new update-credential command is added. This allows locally updated credentials to be uploaded to the controller to replace any existing credentials which may be in use.

### CODE REVIEW

File: cmd/juju/cloud/updatecredential.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	defer client.Close()
+
+	if err := client.UpdateCredential(credentialTag, credToUpdate); err != nil {
+		return c.processError(client, err)
+	}
+	ctx.Infof("Updated credential %q for user %q on cloud %q.", c.credential, accountDetails.User, c.cloud)
+	return nil
+}
+
+func (c *updateCredentialCommand) processError(client credentialAPI, initialErr error) error {
```
Comment (axw) [inline cmd/juju/cloud/updatecredential.go]: -1, this assumes to much. Please either use a special error code or do the check in the client first.

Comment (wallyworld) [inline cmd/juju/cloud/updatecredential.go]: I've moved the check to the backend.

**Comment by wallyworld** [comment]
QA: bootstrap aws, change credential locally, run update-credential. verify also that attempt to update credential on lxd for aws is rejected.

**Comment by wallyworld** [comment]
$$merge$$

**Pull request merged**